sql/core/src/test/scala/org/apache/spark/sql/sources/TableScanSuite.scala

```diff
@@ -315,6 +315,26 @@ class TableScanSuite extends DataSourceTest {
       (1 to 10).map(Row(_)).toSeq)
   }
 
+  test("SPARK-5196 schema field with comment") {
+    sql(
+      """
+       |CREATE TEMPORARY TABLE student(name string comment "SN", age int comment "SA", grade int)
+       |USING org.apache.spark.sql.sources.AllDataTypesScanSource
+       |OPTIONS (
+       |  from '1',
+       |  to '10'
+       |)
+       """.stripMargin)
+
+       val comments = sql("SELECT * FROM student").queryExecution.sparkPlan.
+         schema.fields.map { field =>
```

Comment:
I think it is better to use `queryExecution.executedPlan` since it is the final physical. Also, this line is kind of long. How about we break it as follows.
```
val planned = sql("SELECT * FROM student").queryExecution.executedPlan
val comments = planned.schema.fields.map { ...
```